src/tabs/chat/ChatTabController.tsx

```diff
@@ -0,0 +1,29 @@
+import { TabController } from "../TabController";
+import { ChatTabState, Message } from "./ChatTabState";
+
+export class ChatTabController extends TabController<ChatTabState> {
+    public static Commands = ["say", "whisper"];
+
+    public execute = () => {
+        const command: string = this.model.getState().commandToSend;
+        const m = this.model.getState();
+        const recipient: string = m.selectedRecipient;
+        if (!recipient || recipient === "ALL") {
+            this.model.requestResource({
+                data: "say " + command,
+                method: "POST",
+                resourcePath: ["console"],
+            });
+        } else {
+            const recipientName = recipient.replace("PLAYER_", "");
+            this.model.requestResource({
+                data: `whisper ${recipientName} ${command}`,
+                method: "POST",
+                resourcePath: ["console"],
+            });
```

Comment:
I've noticed in testing that whispering to a player doesn't give any feedback to the sender if on the frontend. I think it would be helpful, and it can be done by updating the state of the messages:
`this.model.update({messages: this.model.getState().messages.concat({type: "CHAT", message: "message"})});`
The in-game console uses the format "You -> {playername}: {message}".